indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -128,15 +133,18 @@ public InputRow get()
     }
   }
 
-  private static void writeBytes(byte[] value, ByteArrayDataOutput out) throws IOException
+  private static void writeBytes(@Nullable byte[] value, ByteArrayDataOutput out) throws IOException
   {
-    WritableUtils.writeVInt(out, value.length);
-    out.write(value, 0, value.length);
+    int length = value == null ? -1 : value.length;
+    WritableUtils.writeVInt(out, length);
+    if (value != null) {
+      out.write(value, 0, value.length);
+    }
   }
 
-  private static void writeString(String value, ByteArrayDataOutput out) throws IOException
+  private static void writeString(@Nullable String value, ByteArrayDataOutput out) throws IOException
   {
-    writeBytes(StringUtils.toUtf8(value), out);
+    writeBytes(StringUtils.toUtf8Nullable(value), out);
```

Comment:
how about making `toUtf8` deal with nulls ? do we really need to have 2 different functions?

Reply:
It makes null control less precise. E. g. would "convert" non-null strings into nullable byte arrays. I want to make nullability problem warnings in IntelliJ errors, and it would require to insert something like `assert != null` or `@SuppressWarnings` or `//noinspection` in such places.